e2e/_suites/fleet/fleet.go

```diff
@@ -537,6 +543,60 @@ func (fts *FleetTestSuite) anAgentIsDeployedToFleetWithInstaller(installerType s
 	return fts.anAgentIsDeployedToFleetWithInstallerAndFleetServer(installerType)
 }
 
+// supported installers: tar, rpm, deb
+func (fts *FleetTestSuite) anAgentIsDeployedToFleetWithInstallerAndTags(installerType string, flags string) error {
+	fts.BeatsProcess = ""
+
+	// FIXME: We need to cleanup the steps to support different operating systems
+	// for now we will force the zip installer type when the agent is running on windows
+	if runtime.GOOS == "windows" && common.Provider == "remote" {
+		installerType = "zip"
+	}
+	fts.ElasticAgentFlags = flags
+	return fts.anAgentIsDeployedToFleetWithInstallerAndFleetServer(installerType)
+}
+
+func (fts *FleetTestSuite) tagsAreInTheElasticAgentIndex() error {
+	var tagsArray []string
+	//ex of flags  "--tag production,linux" or "--tag=production,linux"
+	if fts.ElasticAgentFlags != "" {
+		tags := strings.TrimPrefix(fts.ElasticAgentFlags, "--tag")
+		tags = strings.TrimPrefix(tags, "=")
+		tags = strings.TrimPrefix(tags, " ")
```

Comment:
Do we want to remove the prefix only, or to remove any whitespace?

I can imagine this flag: `--tags="production,       linux"`, and it would be correct, right?

```suggestion
		tags = strings.ReplaceAll(tags, " ", "")
```